webapp-src/src/Profile/App.js

```diff
@@ -64,7 +64,9 @@ class App extends Component {
       } else if (message.type === 'loggedIn') {
         this.setState({loggedIn: message.message}, () => {
           if (!this.state.loggedIn) {
-            this.setState({profileList: false, schemeList: [], curNav: "profile"});
+            this.setState({curNav: "profile"});
```

Comment:
Those 3 `setState` calls should be merged in one, especially since a setState is asynchronous, this could lead to undefined behavior...

Reply:
@babelouest I am unsure about the last of these three `setState` statements (which I will merge into one): is there supposed to be a 1-1 correspondence between `profileList` and `schemeList` members? In other words, when `profileList[0]` logs out and is popped from `profileList`, should `schemeList[0]` also be popped from `schemeList`?

Reply:
If you log out from a component, you should send a message to App component so it will refresh The status of all subcomponents.

Reply:
Right, I understand that @babelouest, in fact I’m working precisely on the message receiver here. I believe that the original code incorrectly zeroes put the entire `profileList` whereas only the last active user should be removed (`profileList[0]`). I’ve checked this part of the fix quite a bit and it looks robust. My question is whether the same fix should be repeated for `schemeList` as well.

Reply:
I think I figured it out: `schemeList` is the current user's list of authentication schemes, so it should be indeed zeroed out each time a user logs out. Declaring this resolved and I assume it will raise a red flag with you if my understanding is incorrect.